system-addon/lib/MessageCenterRouter.jsm

```diff
@@ -132,18 +138,22 @@ class _MessageCenterRouter {
         await this.sendNextMessage(target);
         break;
       case "BLOCK_MESSAGE_BY_ID":
-        await this.setState(state => {
-          const newState = Object.assign({}, state.blockList);
-          newState[action.data.id] = true;
-          return {blockList: newState};
+        await this.setState(async state => {
+          const blockList = state.blockList.slice();
+          blockList.push(action.data.id);
+          await this._storage.set("blockList", blockList);
```

Comment:
I'm not sure we want to wait on the indexedDB to actually set the state here, we probably just want to do that in the background and throw an error if there was a problem with the transaction